File: src/main/java/org/myfintech/payment/PaymentApplication.java

package org.myfintech.payment;

/**
 *
 * @author : Dhanuka Ranasinghe
 * @since : Date: 05/07/2025
 */

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.data.web.config.EnableSpringDataWebSupport;
import org.springframework.scheduling.annotation.EnableAsync;

@EnableAsync
@SpringBootApplication
@EnableSpringDataWebSupport
public class PaymentApplication {
    public static void main(String[] args) {
        SpringApplication.run(PaymentApplication.class, args);
    }
}

File: src/main/kotlin/org/myfintech/payment/PaymentApplication.kt
package org.myfintech.payment

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.data.web.config.EnableSpringDataWebSupport
import org.springframework.scheduling.annotation.EnableAsync

/**
 * Main entry point for the MyFintech Payment Service.
 *
 * @author : Dhanuka Ranasinghe
 * @since : Date: 05/07/2025
 */
@EnableAsync
@SpringBootApplication
@EnableSpringDataWebSupport
class PaymentApplication

/**
 * Top-level main function to bootstrap the Spring Boot application.
 */
fun main(args: Array<String>) {
    runApplication<PaymentApplication>(*args)
}